executor/brie.go

```diff
@@ -265,37 +423,175 @@ func (b *executorBuilder) buildBRIE(s *ast.BRIEStmt, schema *expression.Schema)
 		for _, opt := range s.Options {
 			switch opt.Tp {
 			case ast.BRIEOptionOnline:
 				e.restoreCfg.Online = opt.UintValue != 0
 			}
 		}
+	}
 
-	default:
-		b.err = errors.Errorf("unsupported BRIE statement kind: %s", s.Kind)
+	return e
+}
+
+func (b *executorBuilder) buildImport(s *ast.BRIEStmt, schema *expression.Schema) Executor {
+	e := &BRIEExec{
+		baseExecutor: newBaseExecutor(b.ctx, schema, 0),
+		info: &brieTaskInfo{
+			kind:      s.Kind,
+			originSQL: s.SecureText(),
+		},
+	}
+	tidbCfg := config.GetGlobalConfig()
+
+	var (
+		// IMPORT has two types of config, task config for each import task
+		// global config for lightning as a long-term running server that serve many tasks
+		importGlobalCfg = importcfg.NewGlobalConfig()
+		importTaskCfg   = importcfg.NewConfig()
+	)
+	// set proper values to embedded lightning
+	importGlobalCfg.App.StatusAddr = ":8289"
+	importGlobalCfg.App.Level = tidbCfg.Log.Level
+	importGlobalCfg.Security.CAPath = tidbCfg.Security.ClusterSSLCA
+	importGlobalCfg.Security.CertPath = tidbCfg.Security.ClusterSSLCert
+	importGlobalCfg.Security.KeyPath = tidbCfg.Security.ClusterSSLKey
+	importGlobalCfg.TiDB.Port = int(tidbCfg.Port)
+	importGlobalCfg.TiDB.PdAddr = strings.Split(tidbCfg.Path, ",")[0]
+
+	// set Port and PdAddr to avoid lightning query TiDB HTTP API
+	importTaskCfg.TiDB.Port = int(tidbCfg.Port)
+	importTaskCfg.TiDB.PdAddr = strings.Split(tidbCfg.Path, ",")[0]
+	// StatusPort is not used, but we could pass it to improve robust
+	importTaskCfg.TiDB.StatusPort = int(tidbCfg.Status.StatusPort)
+	importTaskCfg.TikvImporter.Backend = importcfg.BackendLocal
+	importTaskCfg.TikvImporter.SortedKVDir = filepath.Join(tidbCfg.TempStoragePath, defaultImportID)
+	importTaskCfg.Checkpoint.Schema = defaultImportID
+	importTaskCfg.Checkpoint.Driver = importcfg.CheckpointDriverMySQL
+	importTaskCfg.Mydumper.CSV.Header = false // TODO(lance6716): address this behaviour to user
+
+	// handle s.Storage:
+	// IMPORT ... FROM **s.Storage**
+	storageURL, err := url.Parse(s.Storage)
+	if err != nil {
+		b.err = ErrBRIEInvalidExternalStore.GenWithStackByArgs(s.Storage, err)
 		return nil
 	}
+	importTaskCfg.Mydumper.SourceDir = storageURL.String()
+	e.info.storage = storageURL.String()
+
+	// handle s.Tables:
+	// IMPORT **s.Tables** FROM ...
+	switch {
+	case len(s.Tables) != 0:
+		tbls := make([]string, 0, len(s.Tables))
+		for _, tbl := range s.Tables {
+			if tbl.Schema.L == "" {
+				b.err = ErrBRIEInvalidTable.GenWithStackByArgs(fmt.Sprintf("please specify schema for %s in IMPORT", tbl.Name.O))
+				return nil
+			}
+			tbls = append(tbls, fmt.Sprintf("%s.%s", tbl.Schema, tbl.Name))
+		}
+		importTaskCfg.Mydumper.Filter = tbls
+	case len(s.Schemas) != 0:
+		dbs := make([]string, 0, len(s.Schemas))
+		for _, db := range s.Schemas {
+			dbs = append(dbs, fmt.Sprintf("%s.*", db))
+		}
+		importTaskCfg.Mydumper.Filter = dbs
+	default:
+		// no filter means pass all
+	}
```

Comment:
🤔 maybe we should add a TableFilter field in Lightning's config just like BR.

Reply:
comments should have been addressed in https://github.com/pingcap/tidb/pull/21195/commits/f3b6b19bbc0431a5404f6433ef2de4a9ab2f67d3 except for this one. I'll change raw string `Filter` to `filter.Filter` later